From 67e8c93a8652fd31dbc8a5f6c1dbece2f52709a4 Mon Sep 17 00:00:00 2001
From: Timo Aaltonen <tjaalton@debian.org>
Date: Wed, 24 Jan 2024 13:54:37 +0200
Subject: [PATCH] Revert "intel: reduce minimum memory alignment on Gfx12.5"

This reverts commit 9c67dde401ea0ceb0c159428d0775948108343fb.

OEM-6.1 kernel does not have support for this, so stick to the
old alignment requirement until we no longer need to support this
kernel.
---
 src/intel/dev/i915/intel_device_info.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/intel/dev/i915/intel_device_info.c b/src/intel/dev/i915/intel_device_info.c
index cc471f920e6..d2dc1d2a27d 100644
--- a/src/intel/dev/i915/intel_device_info.c
+++ b/src/intel/dev/i915/intel_device_info.c
@@ -607,9 +607,12 @@ bool intel_device_info_i915_get_info_from_fd(int fd, struct intel_device_info *d
    if (getparam(fd, I915_PARAM_HAS_CONTEXT_ISOLATION, &val))
       devinfo->has_context_isolation = val;
 
-   /* TODO: We might be able to reduce alignment to 4Kb on DG1. */
+   /* TODO: i915 don't require anymore the 2Mb alignment for gfx 12.5 and
+    * newer but using 64k brings some issues like unaligned offsets with
+    * aux map aligned to 1Mb in MTL.
+    */
    if (devinfo->verx10 >= 125)
-      devinfo->mem_alignment = 64 * 1024;
+      devinfo->mem_alignment = 2 * 1024 * 1024;
    else if (devinfo->has_local_mem)
       devinfo->mem_alignment = 64 * 1024;
    else
-- 
2.40.1

